### PR TITLE
Make contract clear for KNN nodes

### PR DESCRIPTION
**Pull request by eddyxu**
Fix KNNFlat / KNNIndex node output schema. 
Add preconditions for KNN nodes to check inputs and parameters.
Removed the hack in scanner to pass ann schema explicitly.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -23,9 +23,9 @@ jobs:
           sudo apt install -y protobuf-compiler libssl-dev
       - name: Run cargo fmt
         run: cargo fmt --check
-      - name: Run clippy
+          #- name: Run clippy
```
Comment (changhiskhan) [inline .github/workflows/rust.yml:26]: remove?

File: rust/src/dataset/scanner.rs [hunk 2]
```diff
@@ -406,18 +399,14 @@ impl Scanner {
...
         filter: Arc<dyn PhysicalExpr>,
         input: Arc<dyn ExecutionPlan>,
-        drop_row_id: bool,
-        ann_schema: Option<Arc<Schema>>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
         let filter_node = Arc::new(FilterExec::try_new(filter, input)?);
         let output_schema = self.scanner_output_schema()?;
-        Ok(Arc::new(LocalTakeExec::new(
-            filter_node,
+        Ok(Arc::new(TakeExec::try_new(
```
Comment (changhiskhan) [inline rust/src/dataset/scanner.rs:405]: why did we need a separate LocalTake before?

File: rust/src/index/vector/flat.rs [hunk 3]
```diff
@@ -88,10 +86,8 @@ pub async fn flat_search(
 
             // TODO: use heap
             let indices = sort_to_indices(&scores, None, Some(query.k))?;
-            let batch_with_score = batch.try_with_column(
-                ArrowField::new(SCORE_COLUMN, DataType::Float32, false),
-                scores,
-            )?;
+            let batch_with_score = batch
+                .try_with_column(ArrowField::new(SCORE_COL, DataType::Float32, false), scores)?;
```
Comment (changhiskhan) [inline rust/src/index/vector/flat.rs:90]: what should the score column's field id be? how would we enforce it?

File: rust/src/dataset/scanner.rs (hunk 2 above)
Comment (eddyxu) [reply rust/src/dataset/scanner.rs:405]: Was attempted to make it similar to what C++ does, that it takes the same batch from the input BatchRecords, but later found out it is too difficult to implement that in Rust. In the end, `LocalTake` becomes almost identical to `GlobalTake`.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (eddyxu) [reply .github/workflows/rust.yml:26]: fixed.

File: rust/src/index/vector/flat.rs (hunk 3 above)
Comment (eddyxu) [reply rust/src/index/vector/flat.rs:90]: This is still not clear to me why needs to reinforce column id here. 

Not clear to me what bug it triggers, as all python tests / rust tests passed.  Do you remember which PR # is the fix? i can look into that as well.

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**